Notebooks/SentinelUtilities/SentinelPortal/bookmark_helper.py

```diff
@@ -0,0 +1,318 @@
+                 notes=None,
+                 event_time=None,
+                 query_start_time=None,
+                 query_end_time=None):
+        self.displayName = display_name
+        self.labels = tag_list
+        self.query = query
+        self.queryResult = query_result
+        self.notes = notes
+        self.eventTime = event_time
+        self.queryStartTime = query_start_time
+        self.queryEndTime = query_end_time
+        self.bookmarkId = str(uuid.uuid4())
+
+    @property
+    def bookmarkId(self):
+        """ bookmarkId getter """
+        return self._bookmarkId
+
+    @bookmarkId.setter
+    def bookmarkId(self, bookmark_id):
+        """ bookmarkId setter """
+        self._bookmarkId = bookmark_id
+
+    @property
+    def displayName(self):
+        """ displayName getter """
+        return self._displayName
+
+    @displayName.setter
+    def displayName(self, display_name):
+        """ displayName setter """
+        InputValidation.validate_input('display_name', display_name)
+        self._displayName = display_name
+
+    @property
+    def labels(self):
+        """ labels getter """
+        return self._labels
+
+    @labels.setter
+    def labels(self, tag_list):
+        """ labels setter """
+        if tag_list:
+            self._labels = tag_list
+        else:
+            self._labels = []
+
+    @property
+    def query(self):
+        """ query """
+        return self._query
+
+    @query.setter
+    def query(self, query):
+        """ query setter """
+        InputValidation.validate_input('query', query)
+        self._query = query
+
+    @property
+    def queryResult(self):
+        """ queryResult """
+        return self._queryResult
+
+    @queryResult.setter
+    def queryResult(self, query_result):
+        """ queryResult setter """
+        if query_result:
+            self._queryResult = query_result
+        else:
+            self._queryResult = '{}'
+
+    @property
+    def notes(self):
+        """ notes getter """
+        return self._notes
+
+    @notes.setter
+    def notes(self, notes):
+        """ notes setter """
+        self._notes = notes
+
+    @property
+    def eventTime(self):
+        """ eventTime """
+        return self._eventTime
+
+    @eventTime.setter
+    def eventTime(self, event_time):
+        """ eventTime setter """
+        self._eventTime = event_time
+
+    @property
+    def queryStartTime(self):
+        """ queryStartTime """
+        return self._queryStartTime
+
+    @queryStartTime.setter
+    def queryStartTime(self, query_start_time):
+        """ queryStartTime setter """
+        self._queryStartTime = query_start_time
+
+    @property
+    def queryEndTime(self):
+        """ queryEndTime """
+        return self._queryEndTime
+
+    @queryEndTime.setter
+    def queryEndTime(self, query_end_time):
+        """ queryEndTime setter """
+        self._queryEndTime = query_end_time
+
+class BookmarkModel():
```

Comment:
Can we just call this "Bookmark"?

Reply:
This way I will have the flexibility to validate and modify the values passed in